trace-normalization/benches/normalization_utils.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+
+fn normalize_service_bench(c: &mut Criterion) {
+    let mut group = c.benchmark_group("normalization");
+    let cases = &[
+        ("#test_starting_hash", "test_starting_hash"),
+            ("TestCAPSandSuch", "testcapsandsuch"),
+            (
+                "Test Conversion Of Weird !@#$%^&**() Characters",
+                "test_conversion_of_weird_characters",
+            ),
+            ("$#weird_starting", "weird_starting"),
+            ("allowed:c0l0ns", "allowed:c0l0ns"),
+            ("1love", "love"),
+            ("ünicöde", "ünicöde"),
+            ("ünicöde:metäl", "ünicöde:metäl"),
+            ("Data🐨dog🐶 繋がっ⛰てて", "data_dog_繋がっ_てて"),
+            (" spaces   ", "spaces"),
+            (" #hashtag!@#spaces #__<>#  ", "hashtag_spaces"),
+            (":testing", ":testing"),
+            ("_foo", "foo"),
+            (":::test", ":::test"),
+            ("contiguous_____underscores", "contiguous_underscores"),
+            ("foo_", "foo"),
+            (
+                "\u{017F}odd_\u{017F}case\u{017F}",
+                "\u{017F}odd_\u{017F}case\u{017F}",
+            ), // edge-case
+            ("", ""),
```

Comment:
Empty should default to `unnamed-{language}-service`, if this [RFC](https://github.com/DataDog/architecture/blob/master/rfcs/apm/agent/trace-agent-normalization/rfc.md) still applies to the agent, which I believe does.